runtime/compiler/optimizer/DataAccessAccelerator.cpp

```diff
@@ -102,7 +102,7 @@ int32_t TR_DataAccessAccelerator::perform()
 
        // We cannot handle arraylets because hardware intrinsics act on contiguous memory
        !comp()->generateArraylets() && !TR::Compiler->om.useHybridArraylets() &&
-       (!TR::Compiler->om.isOffHeapAllocationEnabled() || comp()->getOption(TR_DisableVectorBCD)))
+       (!TR::Compiler->om.isOffHeapAllocationEnabled() || !comp()->getOption(TR_DisableVectorBCD)))
```

Comment:
@VermaSh  - Would it be possible to update the condition to make it more clear with the intention here - If offheap is enabled and we do not support VectorBCD then do not do optimization. 
```suggestion
       !(TR::Compiler->om.isOffHeapAllocationEnabled() && comp()->getOption(TR_DisableVectorBCD)))
```

Reply:
I have updated it in my latest update